From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Sat, 2 Nov 2024 06:02:21 +0900
Subject: network/json: add missing entries for route properties

Bug-Ubuntu: https://launchpad.net/bugs/2098183
Origin: backport, https://github.com/systemd/systemd/commit/5ca212a9843440fd15be32068365aa9a9b6d079c
[chengen - replace `SD_*` macros with the original `JSON_*` macros.]

--- systemd-255.4.orig/src/network/networkd-json.c
+++ systemd-255.4/src/network/networkd-json.c
@@ -238,6 +238,7 @@ static int route_append_json(Route *rout
                                                           "Source", JSON_BUILD_IN_ADDR(&route->src, route->family)),
                                 JSON_BUILD_PAIR_UNSIGNED_NON_ZERO("SourcePrefixLength", route->src_prefixlen),
                                 JSON_BUILD_PAIR_IN_ADDR_NON_NULL("PreferredSource", &route->prefsrc, route->family),
+                                JSON_BUILD_PAIR_UNSIGNED("TOS", route->tos),
                                 JSON_BUILD_PAIR_UNSIGNED("Scope", route->scope),
                                 JSON_BUILD_PAIR_STRING("ScopeString", scope),
                                 JSON_BUILD_PAIR_UNSIGNED("Protocol", route->protocol),
@@ -252,6 +253,7 @@ static int route_append_json(Route *rout
                                 JSON_BUILD_PAIR_UNSIGNED("Flags", route->flags),
                                 JSON_BUILD_PAIR_STRING("FlagsString", strempty(flags)),
                                 JSON_BUILD_PAIR_FINITE_USEC("LifetimeUSec", route->lifetime_usec),
+                                JSON_BUILD_PAIR_UNSIGNED_NON_ZERO("NextHopID", route->nexthop_id),
                                 JSON_BUILD_PAIR_STRING("ConfigSource", network_config_source_to_string(route->source)),
                                 JSON_BUILD_PAIR_STRING("ConfigState", state),
                                 JSON_BUILD_PAIR_IN_ADDR_NON_NULL("ConfigProvider", &route->provider, route->family)));
