From: Felipe Sateler <fsateler@gmail.com>
Date: Fri, 24 Jun 2016 22:10:37 -0400
Subject: build-sys: do not recompile everything for libsystemd

---
 Makefile.am | 16 ++++------------
 1 file changed, 4 insertions(+), 12 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index b545cf6..f763532 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3187,16 +3187,7 @@ EXTRA_DIST += \
 	src/libsystemd/sd-bus/DIFFERENCES \
 	src/libsystemd/sd-bus/GVARIANT-SERIALIZATION
 
-libsystemd_la_SOURCES = \
-	$(libsystemd_internal_la_SOURCES) \
-	$(libsystemd_journal_internal_la_SOURCES)
-
-nodist_libsystemd_la_SOURCES = \
-	$(nodist_libsystemd_internal_la_SOURCES)
-
-libsystemd_la_CFLAGS = \
-	$(libsystemd_internal_la_CFLAGS) \
-	$(libsystemd_journal_internal_la_CFLAGS)
+libsystemd_la_SOURCES =
 
 libsystemd_la_LDFLAGS = \
 	$(AM_LDFLAGS) \
@@ -3204,8 +3195,9 @@ libsystemd_la_LDFLAGS = \
 	-Wl,--version-script=$(top_srcdir)/src/libsystemd/libsystemd.sym
 
 libsystemd_la_LIBADD = \
-	$(libsystemd_internal_la_LIBADD) \
-	$(libsystemd_journal_internal_la_LIBADD)
+	libsystemd-internal.la \
+	libbasic.la \
+	libsystemd-journal-internal.la
 
 libsystemd-install-hook:
 	libname=libsystemd.so && $(move-to-rootlibdir)
