Index: systemd/src/cryptsetup/cryptsetup-generator.c
===================================================================
--- systemd.orig/src/cryptsetup/cryptsetup-generator.c
+++ systemd/src/cryptsetup/cryptsetup-generator.c
@@ -370,10 +370,14 @@ static int create_disk(
                         "ExecStartPost=" ROOTLIBEXECDIR "/systemd-makefs ext2 '/dev/mapper/%s'\n",
                         name_escaped);

-        if (swap)
+        if (swap) {
                 fprintf(f,
                         "ExecStartPost=" ROOTLIBEXECDIR "/systemd-makefs swap '/dev/mapper/%s'\n",
                         name_escaped);
+                fprintf(f,
+                        "ExecStartPost=/bin/udevadm trigger '/dev/mapper/%s'\n",
+                        name_escaped);
+        }

         if (keydev)
                 fprintf(f,
