Index: systemd/src/kernel-install/meson.build
===================================================================
--- systemd.orig/src/kernel-install/meson.build
+++ systemd/src/kernel-install/meson.build
@@ -20,14 +20,14 @@ ukify_install = custom_target(
         install_mode : 'rwxr-xr-x',
         install_dir : kernelinstalldir)
 
-loaderentry_install = custom_target(
-        '90-loaderentry.install',
-        input : '90-loaderentry.install.in',
-        output : '90-loaderentry.install',
-        command : [jinja2_cmdline, '@INPUT@', '@OUTPUT@'],
-        install : want_kernel_install,
-        install_mode : 'rwxr-xr-x',
-        install_dir : kernelinstalldir)
+# loaderentry_install = custom_target(
+#         '90-loaderentry.install',
+#         input : '90-loaderentry.install.in',
+#         output : '90-loaderentry.install',
+#         command : [jinja2_cmdline, '@INPUT@', '@OUTPUT@'],
+#         install : want_kernel_install,
+#         install_mode : 'rwxr-xr-x',
+#         install_dir : kernelinstalldir)
 
 uki_copy_install = files('90-uki-copy.install')
 
Index: systemd/test/meson.build
===================================================================
--- systemd.orig/test/meson.build
+++ systemd/test/meson.build
@@ -314,19 +314,19 @@ endif
 
 ############################################################
 
-if want_tests != 'false' and conf.get('ENABLE_KERNEL_INSTALL') == 1
-        kernel_install = executables_by_name.get('kernel-install')
-        args = [kernel_install.full_path(), loaderentry_install.full_path(), uki_copy_install]
-        deps = [kernel_install, loaderentry_install]
-        if want_ukify and boot_stubs.length() > 0
-                args += [ukify.full_path(), ukify_install.full_path(), boot_stubs[0]]
-                deps += [ukify, ukify_install, boot_stubs[0]]
-        endif
-
-        test('test-kernel-install',
-             test_kernel_install_sh,
-             env : test_env,
-             args : args,
-             depends : deps,
-             suite : 'kernel-install')
-endif
+# if want_tests != 'false' and conf.get('ENABLE_KERNEL_INSTALL') == 1
+#         kernel_install = executables_by_name.get('kernel-install')
+#         args = [kernel_install.full_path(), loaderentry_install.full_path(), uki_copy_install]
+#         deps = [kernel_install, loaderentry_install]
+#         if want_ukify and boot_stubs.length() > 0
+#                 args += [ukify.full_path(), ukify_install.full_path(), boot_stubs[0]]
+#                 deps += [ukify, ukify_install, boot_stubs[0]]
+#         endif
+#
+#         test('test-kernel-install',
+#              test_kernel_install_sh,
+#              env : test_env,
+#              args : args,
+#              depends : deps,
+#              suite : 'kernel-install')
+# endif
