From: Lennart Poettering <lennart@poettering.net>
Date: Mon, 21 Jan 2019 18:36:14 +0100
Subject: llmnr: add comment why we install no complete() handler on stream

---
 src/resolve/resolved-llmnr.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/resolve/resolved-llmnr.c b/src/resolve/resolved-llmnr.c
index d73f865..b7c37f1 100644
--- a/src/resolve/resolved-llmnr.c
+++ b/src/resolve/resolved-llmnr.c
@@ -302,6 +302,8 @@ static int on_llmnr_stream(sd_event_source *s, int fd, uint32_t revents, void *u
         }
 
         stream->on_packet = on_llmnr_stream_packet;
+        /* We don't configure a "complete" handler here, we rely on the default handler than simply drops the
+         * reference to the stream, thus freeing it */
         return 0;
 }
 
