From: Nick Rosbrook <enr0n@ubuntu.com>
Date: Thu, 11 Sep 2025 14:55:19 -0400
Subject: test: workaround uutils dd broken pipe

Bug-Ubuntu: https://launchpad.net/bugs/2136752
Forwarded: no

---
 test/units/TEST-72-SYSUPDATE.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/test/units/TEST-72-SYSUPDATE.sh b/test/units/TEST-72-SYSUPDATE.sh
index d74e382..85f7078 100755
--- a/test/units/TEST-72-SYSUPDATE.sh
+++ b/test/units/TEST-72-SYSUPDATE.sh
@@ -21,6 +21,12 @@ if [[ ! -x "$SYSUPDATE" ]]; then
     exit 77
 fi
 
+# Work around broken pipe error in rust-coreutils
+if [[ "$(readlink -f /usr/bin/dd)" == "/usr/lib/cargo/bin/coreutils/dd" ]]; then
+    rm /usr/bin/dd
+    ln -s /usr/bin/gnudd /usr/bin/dd
+fi
+
 # Loopback devices may not be supported. They are used because sfdisk cannot
 # change the sector size of a file, and we want to test both 512 and 4096 byte
 # sectors. If loopback devices are not supported, we can only test one sector
