From: Nick Rosbrook <enr0n@ubuntu.com>
Date: Thu, 10 Aug 2023 11:38:11 -0400
Subject: meson.build: disable -Wformat-overflow on gcc with -O3

Forwarded: no, Ubuntu specific

---
 meson.build | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/meson.build b/meson.build
index 6c50958..e5c0cbf 100644
--- a/meson.build
+++ b/meson.build
@@ -424,6 +424,11 @@ if cc.get_id() == 'gcc'
         possible_common_cc_flags += '-Wno-unused-result'
 endif

+# Disable -Wformat-overflow on gcc with -O3 to prevent false positives.
+if cc.get_id() == 'gcc' and '-O3' in c_args
+        possible_common_cc_flags += '-Wno-format-overflow'
+endif
+
 # --as-needed and --no-undefined are provided by meson by default,
 # run 'meson configure' to see what is enabled
 possible_link_flags = [
