Index: systemd/hwdb.d/60-keyboard.hwdb
===================================================================
--- systemd.orig/hwdb.d/60-keyboard.hwdb
+++ systemd/hwdb.d/60-keyboard.hwdb
@@ -1816,6 +1816,10 @@ evdev:atkbd:dmi:bvn*:bvr*:bd*:svnTIMI:pn
  KEYBOARD_KEY_9c=enter                                  # KP_enter in the main area is wrong
  KEYBOARD_KEY_dd=rightctrl                              # Right Ctrl is preferrable over Menu
 
+# Pangolin 13
+evdev:atkbd:dmi:bvn*:bvr*:bd*:svnSystem76*:pnPangolin*:pvrpang13*
+ KEYBOARD_KEY_76=f21                                    # Touchpad toggle
+
 ###########################################################
 # Samsung
 ###########################################################
