Index: systemd/hwdb.d/60-keyboard.hwdb
===================================================================
--- systemd.orig/hwdb.d/60-keyboard.hwdb
+++ systemd/hwdb.d/60-keyboard.hwdb
@@ -1698,0 +1698,0 @@ evdev:atkbd:dmi:bvn*:bvr*:bd*:svnSystem7
  KEYBOARD_KEY_f7=f21                                    # Touchpad toggle
  KEYBOARD_KEY_76=f21                                    # Touchpad toggle

+# Pangolin 14
+evdev:atkbd:dmi:bvn*:bvr*:bd*:svnSystem76*:pnPangolin*:pvrpang14*
+ KEYBOARD_KEY_76=f21                                    # Touchpad toggle
+
 ###########################################################
 # T-bao
 ###########################################################
