From: Nick Rosbrook <nick.rosbrook@canonical.com>
Date: Thu, 17 Nov 2022 11:29:03 -0500
Subject: test: remove no-longer-needed quirk for set-locale on Debian/Ubuntu

---
 test/units/testsuite-73.sh | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/test/units/testsuite-73.sh b/test/units/testsuite-73.sh
index f9e2dce..1e493c0 100755
--- a/test/units/testsuite-73.sh
+++ b/test/units/testsuite-73.sh
@@ -118,9 +118,7 @@ LC_CTYPE=$i"
 
         assert_rc 0 localectl set-locale "$i"
         if [[ -f /etc/default/locale ]]; then
-            # Debian/Ubuntu patch is buggy, and LC_CTYPE= still exists.
-            assert_eq "$(cat /etc/default/locale)" "LANG=$i
-LC_CTYPE=$i"
+            assert_eq "$(cat /etc/default/locale)" "LANG=$i"
         else
             assert_eq "$(cat /etc/locale.conf)" "LANG=$i"
         fi
