From: Nick Rosbrook <nick.rosbrook@canonical.com>
Date: Wed, 5 Jul 2023 09:59:21 -0400
Subject: test: skip tests for unsupported localectl features

Forwarded: no
---
 test/units/testsuite-73.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/test/units/testsuite-73.sh b/test/units/testsuite-73.sh
index df5af4b..ec3fa6c 100755
--- a/test/units/testsuite-73.sh
+++ b/test/units/testsuite-73.sh
@@ -227,6 +227,9 @@ wait_vconsole_setup() {
 testcase_vc_keymap() {
     local i output vc
 
+    echo "Setting X11 and console keymaps is not supported in Debian, skipping test."
+    return
+
     if [[ -z "$(localectl list-keymaps)" ]]; then
         echo "No vconsole keymap installed, skipping test."
         return
@@ -297,6 +300,9 @@ testcase_vc_keymap() {
 testcase_x11_keymap() {
     local output
 
+    echo "Setting X11 and console keymaps is not supported in Debian, skipping test."
+    return
+
     if [[ -z "$(localectl list-x11-keymap-layouts)" ]]; then
         echo "No x11 keymap installed, skipping test."
         return
