From: Nick Rosbrook <enr0n@ubuntu.com>
Date: Wed, 29 May 2024 12:15:08 -0400
Subject: tmpfiles.d/tmp.conf: make cleanup-age 30d on Ubuntu

Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/2019026
Forwarded: no, 30d cleanup is Ubuntu specific
---
 tmpfiles.d/tmp.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tmpfiles.d/tmp.conf b/tmpfiles.d/tmp.conf
index fe5225d..bce48a0 100644
--- a/tmpfiles.d/tmp.conf
+++ b/tmpfiles.d/tmp.conf
@@ -8,5 +8,5 @@
 # See tmpfiles.d(5) for details
 
 # Clear tmp directories separately, to make them easier to override
-q /tmp 1777 root root 10d
+q /tmp 1777 root root 30d
 q /var/tmp 1777 root root 30d
