From: Lennart Poettering <lennart@poettering.net>
Date: Wed, 15 Jan 2020 17:08:25 +0100
Subject: [PATCH] execute: add const to array parameters, where possible

---
 src/core/execute.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/core/execute.c b/src/core/execute.c
index 4cb59d2..2ab1313 100644
--- a/src/core/execute.c
+++ b/src/core/execute.c
@@ -1125,7 +1125,7 @@ static int setup_pam(
                 gid_t gid,
                 const char *tty,
                 char ***env,
-                int fds[], size_t n_fds) {
+                const int fds[], size_t n_fds) {
 
 #if HAVE_PAM
 
@@ -2805,7 +2805,7 @@ static int close_remaining_fds(
                 int user_lookup_fd,
                 int socket_fd,
                 int exec_fd,
-                int *fds, size_t n_fds) {
+                const int *fds, size_t n_fds) {
 
         size_t n_dont_close = 0;
         int dont_close[n_fds + 12];
